base/reducedim.jl

```diff
@@ -321,21 +332,43 @@ julia> mapreduce(isodd, |, a, dims=1)
 """
 mapreduce(f, op, A::AbstractArrayOrBroadcasted; dims=:, init=_InitialValue()) =
     _mapreduce_dim(f, op, init, A, dims)
-mapreduce(f, op, A::AbstractArrayOrBroadcasted...; kw...) =
-    reduce(op, map(f, A...); kw...)
+
+mapreduce(f, op, A::AbstractArrayOrBroadcasted, Bs::AbstractArrayOrBroadcasted...; dims=:, init=_InitialValue()) =
+    _mapreduce_dim_vararg(f, op, init, (A, Bs...), dims)
 
 _mapreduce_dim(f, op, nt, A::AbstractArrayOrBroadcasted, ::Colon) =
     mapfoldl_impl(f, op, nt, A)
 
 _mapreduce_dim(f, op, ::_InitialValue, A::AbstractArrayOrBroadcasted, ::Colon) =
     _mapreduce(f, op, IndexStyle(A), A)
 
-_mapreduce_dim(f, op, nt, A::AbstractArrayOrBroadcasted, dims) =
-    mapreducedim!(f, op, reducedim_initarray(A, dims, nt), A)
-
 _mapreduce_dim(f, op, ::_InitialValue, A::AbstractArrayOrBroadcasted, dims) =
     mapreducedim!(f, op, reducedim_init(f, op, A, dims), A)
 
+
+_mapreduce_dim_vararg(f, op, init, As::Tuple, ::Colon) =
+    only(_mapreduce_dim_vararg(f, op, init, As, 1:maximum(ndims, As)))
```

Comment:
What I've called `_mapreduce_dim_vararg` could simply be methods of `_mapreduce_dim`. It would be tidiest to re-order its arguments from `_mapreduce_dim(f, op, init, A, dims)` to  `_mapreduce_dim(f, op, init, dims, A, Bs...)`. Would this be acceptable?

According to https://juliahub.com/ui/Search?q=_mapreduce_dim&type=code there are 3 packages which overload `_mapreduce_dim`, so perhaps it's not a disaster to change it.